etc/afpd/status.c

```diff
@@ -671,7 +671,11 @@ void set_signature(struct afp_options *options) {
                     goto server_signature_random;
                 } else {
                     fseek(fp, -1L, SEEK_END);
-                    if(fgetc(fp) != '\n') fputc('\n', fp); /* last char is \n? */
+                    if(fgetc(fp) != '\n')
+                    {
+                        fflush(fp);
+                        fputc('\n', fp); /* last char is \n? */
```

Comment:
Why? This doesn't seem to make sense.

Reply:
As I understand it, this is in order to flush the bidirectional i/o stream when switching direction.

Reply:
I had a chat with Chris. We would suggest using fseek here instead for C11 conformance. Please me know if this looks good to you now. I'll make a clean PR.

Reply:
Ah, that one, now I remember. So either please squash the two commits or drop the last one.

Reply:
New PR in https://github.com/Netatalk/Netatalk/pull/168